### PR TITLE
[cnpy-1.0] upload lib

### PR DESCRIPTION
**Pull request by hongyx11**
Specify library name and version: cnpy/1.0

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/cnpy/all/test_package/conanfile.py
Comment (madebr) [inline recipes/cnpy/all/test_package/conanfile.py:10]:
```suggestion

```

File: recipes/cnpy/all/conanfile.py
Comment (madebr) [inline recipes/cnpy/all/conanfile.py:39]:
```suggestion
    def configure(self):
        if self.options.shared:
            del self.options.fPIC
        del self.settings.compiler.libcxx
```

File: recipes/cnpy/all/conanfile.py
Comment (madebr) [inline recipes/cnpy/all/conanfile.py:42]:
```suggestion
        del self.settings.compiler.cppstd

    def requirements(self):
        self.requires("zlib/1.2.11")

    def source(self):
```

File: recipes/cnpy/all/conanfile.py
Comment (madebr) [inline recipes/cnpy/all/conanfile.py:65]:
```suggestion
        self.cpp_info.libs = ["cnpy"]
```

File: recipes/cnpy/all/conanfile.py
Comment (madebr) [inline recipes/cnpy/all/conanfile.py:7]:
```suggestion
    description = "library to read/write .npy and .npz files in C/C++"
```

File: recipes/cnpy/all/test_package/CMakeLists.txt
Comment (madebr) [inline recipes/cnpy/all/test_package/CMakeLists.txt:8]:
```suggestion
add_executable(${PROJECT_NAME} example1.cpp)
target_link_libraries(${PROJECT_NAME} ${CONAN_LIBS})
set_property(TARGET ${PROJECT_NAME} PROPERTY CXX_STANDARD 11)
```

File: recipes/cnpy/all/conanfile.py
Comment (uilianries) [inline recipes/cnpy/all/conanfile.py:8]:
```suggestion
    license = "MIT"
```

**Review by hongyx11 — COMMENTED** [review]
accept change of cxx standard and update zlib requirements, but since this lib needs zlib,  I still need to update dependency in the source code. I tried to use conanfile.py for installation, but I met a problem: how to init conan env using conanfile.py?

**Comment by uilianries** [comment]
@hongyx11 Could you explain why you are packaging your fork and not the original package? 

The original project should be the official package. However, if it's abandoned a fork can be adopted, but it should be accept as "official" fork by the community.

**Comment by hongyx11** [comment]
@uilianries thanks for your message, the reason is I don't how to retrive the source from a github repo. all the examples I saw in conan center are using tar file/ zip file to retrive source. The original repo doesn't have a tar file (tag/release). Please tell me how to do it if you know, thanks! I can change the fork repo to original repo.

**Comment by uilianries** [comment]
@hongyx11 PLease, take a look: https://github.com/hongyx11/conan-center-index/pull/1

File: recipes/cnpy/all/conanfile.py
Comment (prince-chrismc) [inline recipes/cnpy/all/conanfile.py:43]: seems to be a C++ project https://github.com/rogersce/cnpy/blob/4e8810b1a8637695171ed346ce68f6984e585ef4/CMakeLists.txt#L8

```suggestion
```

File: recipes/cnpy/all/test_package/conanfile.py
Comment (uilianries) [inline recipes/cnpy/all/test_package/conanfile.py:17]:
```suggestion

    def build(self):
        cmake = CMake(self)
        cmake.configure()
        cmake.build()

    def test(self):
        if not tools.cross_building(self.settings):
            bin_path = os.path.join("bin", "test_package")
            self.run(bin_path, run_environment=True)
```

Keep it as simple as possible. Don't include new requirements, the idea here is validating the package, not running test packages.

File: recipes/cnpy/all/test_package/example1.cpp
Comment (uilianries) [inline recipes/cnpy/all/test_package/example1.cpp:67]:
```suggestion
#include <cstdlib>
#include<vector>
#include<complex>

#include"cnpy.h"

int main() {
    const int Nx = 128;
    const int Ny = 64;
    const int Nz = 32;

    std::vector<std::complex<double>> data(Nx*Ny*Nz);
    for(int i = 0;i < Nx*Ny*Nz;i++) data[i] = std::complex<double>(rand(),rand());

    cnpy::npy_save("arr1.npy",&data[0],{Nz,Ny,Nx},"w");
    cnpy::NpyArray arr = cnpy::npy_load("arr1.npy");
    std::complex<double>* loaded_data = arr.data<std::complex<double>>();

    return EXIT_SUCCESS;
}
```

**Review by uilianries — CHANGES_REQUESTED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Review by madebr — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Comment by madebr** [comment]
@hongyx11 
I think you accidentally closed this pr.
It got the required approvals and was about to be merged.

**Comment by hongyx11** [comment]
> @hongyx11
> I think you accidentally closed this pr.
> It got the required approvals and was about to be merged.

sorry for the misleading close, I reopen it, thanks!

**Comment by prince-chrismc** [comment]
@uilianries is blocking the PR, when he's back to work on Monday I'm sure he'll update his review

From prince-chrismc/conan-center-index-pending-review#1

PR | By | Recipe | Reviews |  Blockers |  Approvers 
-- | -- | -- | -- | -- | --
#4032 | hongyx11 | cnpy | 11 | uilianries | madebr, prince-chrismc, SSE4

**Review by uilianries — APPROVED** [review]
LGTM

**Pull request merged**
